addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -141,28 +141,20 @@ def _get_l10n_latam_documents_domain(self):
             internal_types = ['invoice', 'debit_note']
         return [('internal_type', 'in', internal_types), ('country_id', '=', self.company_id.country_id.id)]
 
-    @api.depends('journal_id', 'partner_id', 'company_id')
-    def _compute_l10n_latam_documents(self):
+    @api.depends('journal_id', 'partner_id', 'company_id', 'type')
+    def _compute_l10n_latam_available_document_types(self):
+        self.l10n_latam_available_document_type_ids = False
+        for rec in self.filtered(lambda x: x.journal_id and x.l10n_latam_use_documents and x.partner_id):
+            rec.l10n_latam_available_document_type_ids = self.env['l10n_latam.document.type'].search(rec._get_l10n_latam_documents_domain())
+
+    @api.depends('l10n_latam_available_document_type_ids')
+    @api.depends_context('internal_type')
+    def _compute_l10n_latam_document_type(self):
         internal_type = self._context.get('internal_type', False)
-        recs_with_journal_partner = self.filtered(lambda x: x.journal_id and x.l10n_latam_use_documents and x.partner_id)
-        for rec in recs_with_journal_partner:
-            document_types = self.env['l10n_latam.document.type'].search(rec._get_l10n_latam_documents_domain())
-
-            # If internal_type is in context we try to search for an specific document. for eg used on debit notes
-            document_type = internal_type and document_types.filtered(
-                lambda x: x.internal_type == internal_type) or document_types
-
-            rec.l10n_latam_available_document_type_ids = document_types
-            rec.l10n_latam_document_type_id = document_type and document_type[0]
-        remaining = self - recs_with_journal_partner
-        remaining.l10n_latam_available_document_type_ids = []
-        remaining.l10n_latam_document_type_id = False
-
-    @api.model_create_multi
-    def create(self, vals_list):
-        recs = super().create(vals_list)
-        recs.filtered(lambda x: x.l10n_latam_use_documents and not x.l10n_latam_document_type_id)._compute_l10n_latam_documents()
-        return recs
+        for rec in self.filtered(lambda x: x.state == 'draft' and x.l10n_latam_available_document_type_ids):
+            document_types = rec.l10n_latam_available_document_type_ids._origin
```

Comment:
@zaoral Maybe I am missing something, but why do we need ._origin?

Reply:
This was the recommendation of @rco-odoo because it was returning new recs and it was not working without it.
He told me that possible, in the future, it was going to change and that it wouldn't be necessary anymore.